dist/common/scripts/scylla_io_setup

```diff
@@ -88,56 +142,42 @@ if __name__ == "__main__":
-            if not "data_file_directories" in cfg:
-                cfg["data_file_directories"] = [os.path.join(default_path, 'data')]
-            data_dirs = cfg["data_file_directories"]
-
-            for t in [ "commitlog", "hints", "view_hints", "saved_caches" ]:
-                key = "%s_directory" % t
-                if key in cfg:
-                    data_dirs += [ cfg[key] ]
-                elif os.path.isdir(os.path.join(default_path, t)):
-                    data_dirs += [ os.path.join(default_path, t) ]
-
-            iotune_args = []
-            for data_dir in data_dirs:
-                if os.path.exists(data_dir) == False:
-                    logging.error("%s was not found. Please check the configuration and run scylla_io_setup again.\n", data_dir)
-                    sys.exit(1)
-                if os.path.isdir(data_dir) == False:
-                    logging.error("%s is not a directory. Please check the configuration and run scylla_io_setup again.\n", data_dir)
-                    sys.exit(1)
-                st = os.statvfs(data_dir)
-                avail = st.f_bavail * st.f_frsize
-                rec = 10000000000
-                if avail < rec:
-                    logging.error("Filesystem at %s has only %d bytes available; that is less than the recommended 10 GB. Please free up space and run scylla_io_setup again.\n", data_dir, avail)
-                    sys.exit(1)
-                blocktune.tune_fs(data_dir, '2')
-                iotune_args += [ "--evaluation-directory", data_dir ]
-
-            if cpudata.cpuset():
-                iotune_args += [ "--cpuset", ",".join(map(str, cpudata.cpuset())) ]
-            elif cpudata.smp():
-                iotune_args += [ "--smp", str(cpudata.smp()) ]
-
-            try:
-                subprocess.check_call([bindir() + "/iotune",
-                                       "--format", "envfile",
-                                       "--options-file", etcdir() + "/scylla.d/io.conf",
-                                       "--properties-file", etcdir() + "/scylla.d/io_properties.yaml"] + iotune_args)
-            except Exception:
-                logging.error("%s did not pass validation tests, it may not be on XFS and/or has limited disk space.\n"
-                              "This is a non-supported setup, and performance is expected to be very bad.\n"
-                              "For better performance, placing your data on XFS-formatted directories is required.\n"
-                              "To override this error, enable developer mode as follow:\n"
-                              "sudo %s/scylla_dev_mode_setup --developer-mode 1", data_dirs, scriptsdir())
-                sys.exit(1)
+        elif gcp_instance().is_gce_instance() and (is_debian_variant() or is_redhat_variant()):
+            idata = gcp_instance()
+            idata.populate()  # lazy populate, avoid curl calls in case it's not gce_instance
```

Comment:
I don't think what populate() does is what's called "lazy".
It would be lazy if you wouldn't read info until you needed it (try to read it) which doesn't seem to be the case.

Reply:
fixed

Reply:
fixed